packages/@ionic/cli/src/lib/hooks.ts

```diff
@@ -97,6 +104,23 @@ export abstract class Hook {
 
     debug(`Could not load hook function ${strong(p)}: %o not a function`, module);
   }
+
+  private generateCTXEnvironment(input: HookInput): NodeJS.ProcessEnv {
+    let environment: NodeJS.ProcessEnv = {};
+
+    for (const [key, value] of Object.entries(input)) {
+      if (typeof value === 'object') {
+        environment = {
+          ...environment,
+          ...this.generateCTXEnvironment(value),
+        };
+      } else {
+        environment[ 'IONIC_CLI_HOOK_CTX_' + key.toUpperCase() ] = value;
```

Comment:
`key` might have weird characters, I suggest this:

```suggestion
        environment[`IONIC_CLI_HOOK_CTX_${lodash.snakeCase(key.toUpperCase())}`] = value;
```

Reply:
That's definitely better

Reply:
Changed it to
``` javascript
const name = [path, key].join('_');

environment[`IONIC_CLI_HOOK_CTX_${lodash.snakeCase(name)}`.toUpperCase()] = value;
```

Output now looks like:
``` bash
IONIC_CLI_HOOK_CTX_NAME=capacitor:run
IONIC_CLI_HOOK_CTX_CAPACITOR_APP_ID=io.ionic.starter
IONIC_CLI_HOOK_CTX_CAPACITOR_APP_NAME=test-cli-app
IONIC_CLI_HOOK_CTX_CAPACITOR_SERVER_URL=http://localhost:8100
IONIC_CLI_HOOK_CTX_CAPACITOR_VERBOSE=false
IONIC_CLI_HOOK_CTX_SERVE_ENGINE=browser
IONIC_CLI_HOOK_CTX_SERVE_EXTERNAL_ADDRESS_REQUIRED=false
IONIC_CLI_HOOK_CTX_SERVE_HOST=localhost
IONIC_CLI_HOOK_CTX_SERVE_LAB=false
IONIC_CLI_HOOK_CTX_SERVE_LAB_HOST=localhost
IONIC_CLI_HOOK_CTX_SERVE_LAB_PORT=8200
IONIC_CLI_HOOK_CTX_SERVE_LIVERELOAD=true
IONIC_CLI_HOOK_CTX_SERVE_OPEN=false
IONIC_CLI_HOOK_CTX_SERVE_PORT=8100
IONIC_CLI_HOOK_CTX_SERVE_PROJECT=app
IONIC_CLI_HOOK_CTX_SERVE_PROXY=true
IONIC_CLI_HOOK_CTX_SERVE_VERBOSE=false
```